consensus/istanbul/backend/engine.go

```diff
@@ -452,12 +452,13 @@ func (sb *backend) Finalize(chain consensus.ChainReader, header *types.Header, s
 		return nil, consensus.ErrInvalidBaseFee
 	}
 
+	var spec *reward.RewardSpec
+	var err error
```

Comment:
@henry-will  Deleting the line will look like this, and we'll get errors:

```go
var spec *reward.RewardSpec
if condition {
    rewardSpec, err = reward.CalcDeferredReward(header, chain.Config())  // undefined: err
} else {
    rewardSpec, err = reward.CalcDeferredRewardSimple(header, chain.Config()) // undefined: err
}
```

However, we can't use `:=` as below because inner-scope and outer-scope rewardSpec are different.
```go
var rewardSpec *reward.RewardSpec

if condition {
    rewardSpec, err := reward.CalcDeferredReward(header, chain.Config()) // this rewardSpec is different from outer rewardSpec!
} else {
    rewardSpec, err := reward.CalcDeferredRewardSimple(header, chain.Config()) // this rewardSpec is different from outer rewardSpec!
}

reward.DistributeBlockReward(state, rewardSpec.Rewards) // outer rewardSpec is uninitialized
```

Reply:
You can also check https://stackoverflow.com/questions/23193208/how-to-initialize-error-type-in-if-else
I'm not sure how we can further optimize it

Reply:
I understand that. thank you